proto/api/v1/cloud_source_service.proto

```diff
@@ -0,0 +1,155 @@
+syntax = "proto3";
+
+option go_package = "v1";
+option java_package = "io.stackrox.proto.api.v1";
+
+import weak "google/api/annotations.proto";
+import "api/v1/empty.proto";
+import "api/v1/pagination.proto";
+import "gogoproto/gogo.proto";
+
+package v1;
+
+// CloudSource is an integration which provides a source for discovered
+// clusters.
+message CloudSource {
+    string id                            = 1;
+    string name                          = 2;
+    string type                          = 3;
+    bool   skip_test_integration         = 4;
+
+    oneof Config {
+        PaladinCloudConfig paladin_cloud = 5;
+        OCMConfig          ocm           = 6;
+    }
+}
+
+// PaladinCloudConfig provides information required to fetch discovered
+// clusters from Paladin Cloud.
+message PaladinCloudConfig {
+    string endpoint = 1 [(gogoproto.moretags) = 'scrub:"dependent" validate:"nolocalendpoint"'];
+    string token    = 2 [(gogoproto.moretags) = 'scrub:"always"'];
+}
+
+// OCMConfig provides information required to fetch discovered clusters from
+// the OpenShift cluster manager.
+message OCMConfig {
+    string endpoint = 1 [(gogoproto.moretags) = 'scrub:"dependent" validate:"nolocalendpoint"'];
+    string token    = 2 [(gogoproto.moretags) = 'scrub:"always"'];
+}
+
+message CloudSourcesFilter {
+    // Matches cloud sources based on their type.
+    repeated string types = 1;
+}
+
+message CountCloudSourcesRequest {
+    // Filters cloud sources based on the requested fields.
+    CloudSourcesFilter filter = 1;
+}
+
+message CountCloudSourcesResponse {
+    int32 count = 1;
+}
+
+message GetCloudSourceRequest {
+    string id = 1;
+}
+
+message GetCloudSourceResponse {
+    CloudSource item = 1;
```

Comment:
Do we want to name this "item" or rather `cloud_source`? Given the other APIs we have, we typically use the name of the message instead of something generic such as "item".